### PR TITLE
resolves #1297 Studio: First implementation of a CAS authentication

### LINKED ISSUE #1297: [Studio] Prise en charge de l'authentification via CAS
Mise en place de la prise en charge du protocole CAS pour l'authentification pour l'accès au Studio Tock

Dans le cadre de notre roadmap, nous envisageons de donner accès à des utilisateurs métiers au Studio afin qu'ils puissent gérer en autonomie l'entrainement et le contenu de leurs bots.
Pour le faire nous devons nous brancher au système d'habilitation Arkea basé sur le protocole CAS afin de bénéficier du SSO

Note : Backlog Arkea

### PR DESCRIPTION
**Pull request by user20339**
# CAS Authentification module for Tock

See #1297 

## New
- Abstract class to be inherited from
  - Extends `SSOTockAuthProvider`
- Additional admin documentation
- Sample code to creates a custom CAS module

## Changes
- Plug CAS in WebVerticle using a Tock property
-  Enforce `Content-Type` of index.html (fix weird issue)
- `PAC4J` in section `<dependencyManagement>` of root `pom.xml`

### CODE REVIEW

File: samples/tock-sample-cas-auth-provider/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Now using version control
+cd my-company-cas
+git init
+```
+
+Using JDK 8+
+```shell
+mvn clean install
+```
+
```
Comment (djowalker) [inline samples/tock-sample-cas-auth-provider/README.md:34]: Add information to be careful to the path in `<relativePath>../../tock</relativePath>` in pom.xml according to where the user installed tock.

File: samples/tock-sample-cas-auth-provider/conf/dev/tock-bot-admin-server.env [hunk 2]
```diff
@@ -0,0 +1,6 @@
+tock_cas_auth_proxy_host=127.0.0.1
+tock_cas_auth_proxy_port=3128
+tock_env=integ
+tock_https_env=false
+botadminverticle_content_path=bot/admin/web/dist
```
[the remark this answers is not in the document]
Comment (djowalker) [reply samples/tock-sample-cas-auth-provider/conf/dev/tock-bot-admin-server.env:5]: This actual path is the one useful when doing `npm install`

[the remark this answers is not in the document]
Comment (user20339) [reply samples/tock-sample-cas-auth-provider/conf/dev/tock-bot-admin-server.env]: done

**Review by djowalker — APPROVED** [review]
Seen, now waiting for an eventual look from @vsct-jburet or/and @francoisno or someone else

**Comment by Benvii** [comment]
@francoisno or @vsct-jburet can you take some time to review it ? Thanks in advance for your review

At CM Arkea we use a CAS for internal authentication, I'm pretty sure that other TOCK users will also benefit from this implementation.

**Comment by djowalker** [comment]
@francoisno @vsct-jburet For the CAS sample, we can't create the repository to publish the sample.
Could you create it in order to move the tock-sample-cas-auth-provier and its README?

**Comment by vsct-jburet** [comment]
@djowalker done https://github.com/theopenconversationkit/tock-sample-cas-auth-provider

**Comment by vsct-jburet** [comment]
and sorry for the delay

**Review by vsct-jburet — APPROVED** [review]

**Comment by vsct-jburet** [comment]
Thank you for the PR. I fix some typo and I merge.

**Comment by vsct-jburet** [comment]
merged

**Comment by djowalker** [comment]
> @djowalker done https://github.com/theopenconversationkit/tock-sample-cas-auth-provider

@vsct-jburet i have a 404, i don't see the repository, did you forgot to push ?

**Comment by djowalker** [comment]
@vsct-jburet still not available to move the code in https://github.com/theopenconversationkit/tock-sample-cas-auth-provider

**Comment by djowalker** [comment]
@francoisno @vsct-jburet still not available to move the code in https://github.com/theopenconversationkit/tock-sample-cas-auth-provider

**Pull request closed without merging**